stratum/src/lib.rs

```diff
@@ -83,18 +87,37 @@ impl Stratum {
 			dispatcher: dispatcher,
 			workers: Arc::new(RwLock::new(HashMap::new())),
 			secret: secret,
+			notify_counter: RwLock::new(NOTIFY_CONTER_INITIAL),
 		});
 
 		let mut delegate = IoDelegate::<Stratum>::new(stratum.clone());
-		delegate.add_method("miner.subscribe", Stratum::subscribe);
-		delegate.add_method("miner.authorize", Stratum::authorize);
+		delegate.add_method("mining.subscribe", Stratum::subscribe);
+		delegate.add_method("mining.authorize", Stratum::authorize);
+		delegate.add_method("mining.submit", Stratum::submit);
 		stratum.handler.add_delegate(delegate);
 
 		try!(stratum.rpc_server.run_async());
 
 		Ok(stratum)
 	}
 
+	fn submit(&self, params: Params) -> std::result::Result<jsonrpc_core::Value, jsonrpc_core::Error> {
+		Ok(match params {
+			Params::Array(vals) => {
+				// first two elements are service messages (worker_id & job_id)
+				self.dispatcher.submit(vals.iter().skip(2)
```

Comment:
There are many places inside dispatch that can lead to work package being unaccepted. Why those errors are not propagated to RPC API user? We're silently failing (unless you run with proper logs) - I think at least returning `false` here could help with debugging in a future.